touchdown/ssh/client.py

```diff
@@ -96,6 +120,15 @@ def verify_transport(self):
         # but rubbish for developers D:
         return
 
+    def get_input_encoding(self):
+        result_buf = six.StringIO()
+        transport = self.get_transport()
+        self._run(transport, 'printenv LANG', stdout=result_buf)
+        result_buf.seek(0)
+        lang = result_buf.read()
+        if '.' in lang:
+            self.input_encoding = lang.rsplit('.', 1)[-1]
+
```

Comment:
Yup. All this encoding stuff is a bit circumspect as we can't guarantee
that output from commands run is intended to be interpreted as text or not.
That's why subprocess doesn't decode by default. I wonder if we could write
the output bytes directly to stdout...
On 17 Mar 2015 08:46, "Jc2k" <notifications@github.com> wrote:

> In touchdown/ssh/client.py
> <https://github.com/yaybu/touchdown/pull/14#discussion_r26556460>:
>
> > @@ -96,6 +120,15 @@ def verify_transport(self):
> >          # but rubbish for developers D:
> >          return
> >
> > +    def get_input_encoding(self):
> > +        result_buf = six.StringIO()
> > +        transport = self.get_transport()
> > +        self._run(transport, 'printenv LANG', stdout=result_buf)
> > +        result_buf.seek(0)
> > +        lang = result_buf.read()
> > +        if '.' in lang:
> > +            self.input_encoding = lang.rsplit('.', 1)[-1]
> > +
>
> I was wondering if there was a better paramiko API for this (our _run
> method is meant for streaming logs as they happen rather than being used
> like subprocess's check_output), alas the exec_command method looks a bit
> rubbish. So let's keep it like this [image: :+1:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/yaybu/touchdown/pull/14/files#r26556460>.
>